### PR TITLE
humanized commands

### PR DESCRIPTION
**Pull request by rangelrey**

### CODE REVIEW

File: hot_tweets_bot/utils.py [hunk 1]
```diff
@@ -32,3 +32,12 @@ def _wrapped(*args, **kwargs):
...
     return _wrapper
+
+
+def create_regex(input):
+    regex_start = '\\w*(?i)'
+    regex_piece = 'x(\\s*|_*|-*)'
+    regex_body = ''
+    for c in list(input.replace(" ", "").replace("_", "").replace("-", "")):
+        regex_body += regex_piece.replace('x', c)
+    return regex_start + regex_body
```
Comment (rooom13) [inline hot_tweets_bot/utils.py:43]: 1. delete lines 38 & 39 & chande 43 with:

`return f"\\w*(?i){regex_body}"`

2. add a docstring documenting what the function does +  what the regex is about
3. create_regex func name is too general, what kind of regex are you creating? Name shoudl be self explanatory
4. regex_piece is not self explanatory, what is that piece of regex doing?
5. in 41: a string is already a iterable, no need to convert it to list()

File: hot_tweets_bot/app.py [hunk 2]
```diff
@@ -362,11 +362,15 @@ def run(self):  # pragma: no cover
             entry_points=[
                 CommandHandler("start", self.callback_on_start),
                 CallbackQueryHandler(self.callback_on_start, pattern=r"\/start"),
+                MessageHandler(Filters.regex(utils.create_regex("start")), self.callback_on_start),
                 CommandHandler("help", self.callback_on_help),
                 CallbackQueryHandler(self.callback_on_help, pattern=r"\/help"),
+                MessageHandler(Filters.regex(utils.create_regex("help")), self.callback_on_help),
                 CommandHandler("my_searches", self.callback_on_my_searches),
                 CallbackQueryHandler(self.callback_on_my_searches, pattern=r"\/my_searches"),
+                MessageHandler(Filters.regex(utils.create_regex("my searches")), self.callback_on_my_searches),
```
Comment (rooom13) [inline hot_tweets_bot/app.py:371]: let's pass the command itself my_searches.

You can put it in a variable before the commandhandler & callback query handler: e.g:

command = "my_searches"
CommandHandler(command, self.callback_on_my_searches),
CallbackQueryHandler(self.callback_on_my_searches, pattern=fr"\/{command"),
                MessageHandler(Filters.regex(utils.create_regex("command"))

File: tests/test_utils.py [hunk 3]
```diff
@@ -10,3 +10,11 @@ def func(a):
     func(None)
     time.sleep(0.02)
     func(None)
+
+
+def test_create_regex():
```
Comment (rooom13) [inline tests/test_utils.py:15]: this test testes the function but not the regex

Please make use of a @pytest.mark.parametrize decorator for testing all test cases

**Review by rooom13 — CHANGES_REQUESTED** [review]

File: hot_tweets_bot/app.py [hunk 4]
```diff
@@ -64,16 +64,31 @@ class States:
...
+command_start = "start"
+command_help = "help"
+command_stop = "stop"
+command_delete_search = "delete_search"
+command_info = "info"
+command_users = "users"
+command_top_tweets = "top_tweets"
+command_backup = "backup"
+command_stop_feedback = "stop_feedback"
+command_admin = "admin"
```
Comment (rooom13) [inline hot_tweets_bot/app.py:79]: pls put this in a class so they are not just global variables

```
class Commands:
    add_search = "..."
    ...
```

File: hot_tweets_bot/utils.py [hunk 5]
```diff
@@ -32,3 +32,12 @@ def _wrapped(*args, **kwargs):
             return f(*args, **kwargs)
         return _wrapped
     return _wrapper
+
+
+def create_regex_contains_characters(input):
```
Comment (rooom13) [inline hot_tweets_bot/utils.py:37]: please add type hints, i.e.: `def create_regex_contains_characters(input: str) -> str:`

File: tests/test_utils.py [hunk 6]
```diff
@@ -10,3 +10,12 @@ def func(a):
     func(None)
     time.sleep(0.02)
     func(None)
+
+
+def test_create_regex_contains_characters():
```
Comment (rooom13) [inline tests/test_utils.py:15]: 1. consider using `@pytest.mark.parametrize` ->check this article: https://docs.pytest.org/en/6.2.x/example/parametrize.html#different-options-for-test-ids  You can as well find some examples within the code

2. you are testing the regex creation but not the regex itself. Please test it so we are sure the regex works as expected

**Review by rooom13 — CHANGES_REQUESTED** [review]

File: hot_tweets_bot/app.py [hunk 7]
```diff
@@ -360,38 +376,44 @@ def run(self):  # pragma: no cover
...
-                CommandHandler("start", self.callback_on_start),
-                CallbackQueryHandler(self.callback_on_start, pattern=r"\/start"),
-                CommandHandler("help", self.callback_on_help),
-                CallbackQueryHandler(self.callback_on_help, pattern=r"\/help"),
-                CommandHandler("my_searches", self.callback_on_my_searches),
-                CallbackQueryHandler(self.callback_on_my_searches, pattern=r"\/my_searches"),
-                CallbackQueryHandler(self.callback_inline_delete_search, pattern=r"\/delete_search .*"),
+                CommandHandler(Commands.start, self.callback_on_start),
+                CallbackQueryHandler(self.callback_on_start, pattern=rf"\/{Commands.start}"),
+                MessageHandler(Filters.regex(utils.create_regex_contains_characters(f"/{Commands.start}")), self.callback_on_start),
```
Comment (rooom13) [inline hot_tweets_bot/app.py:381]: porqué a la regex se le pasa /start y no directamente start? Mejor pasarle start directamente

File: hot_tweets_bot/utils.py [hunk 8]
```diff
@@ -32,3 +32,12 @@ def _wrapped(*args, **kwargs):
             return f(*args, **kwargs)
         return _wrapped
     return _wrapper
+
+
+def create_regex_contains_characters(input: str) -> str:
+    "Creates a regex that matches strings that contain a particular text ignoring split characters '_', '-', ' ','/"
+    regex_ignore_split_characters = "x(\\s*|x*|_*|-*)"
```
Comment (rooom13) [inline hot_tweets_bot/utils.py:39]:
```
a = "pepe{char}"
a.format(char="churra") 
>>> "pepechurra"

```

File: hot_tweets_bot/utils.py [hunk 9]
```diff
@@ -32,3 +32,12 @@ def _wrapped(*args, **kwargs):
...
+    regex_body = ""
```
Comment (rooom13) [inline hot_tweets_bot/utils.py:40]: you can as well first define all chars to remove

```
rem_chars = "_- /"
for c in rem_chars
    input = input.replace(c, "")
```

In this way you can maybe generate the var `regex_ignore_split_characters` based on rem_chars

File: tests/test_utils.py [hunk 10]
```diff
@@ -10,3 +12,33 @@ def func(a):
...
+    ('start', 'start', True),
+    ('start', 'stlart', False),
+    ('please start', 'stlart', False)
+])
+def test_regex(a, b, expected):
+    if re.match(utils.create_regex_contains_characters(a), b):
+        result = True
+    else:
+        result = False
+    assert result == expected
```
Comment (rooom13) [inline tests/test_utils.py:44]: nice usage of test parametrization.

**Review by rooom13 — CHANGES_REQUESTED** [review]

File: hot_tweets_bot/app.py [hunk 11]
```diff
@@ -360,38 +376,44 @@ def run(self):  # pragma: no cover
...
+                CallbackQueryHandler(self.callback_on_start, pattern=Commands.start),
+                MessageHandler(Filters.regex(utils.create_regex_contains_characters(Commands.start)), self.callback_on_start),
+                CommandHandler(Commands.help, self.callback_on_help),
+                CallbackQueryHandler(self.callback_on_help, pattern=Commands.help),
+                MessageHandler(Filters.regex(utils.create_regex_contains_characters(Commands.help)), self.callback_on_help),
+                CommandHandler(Commands.my_searches, self.callback_on_my_searches),
+                CallbackQueryHandler(self.callback_on_my_searches, pattern=Commands.my_searches),
+                MessageHandler(Filters.regex(utils.create_regex_contains_characters(Commands.my_searche)), self.callback_on_my_searches),
+                CallbackQueryHandler(self.callback_inline_delete_search, pattern=rf"{Commands.delete_search} .*"),
+                MessageHandler(Filters.regex(utils.create_regex_contains_characters(Commands.delete_searc)), self.callback_inline_delete_search),
```
Comment (rooom13) [inline hot_tweets_bot/app.py:389]: What is this `Commands.delete_searc`, `Commands.my_searche)`

File: hot_tweets_bot/app.py [hunk 12]
```diff
@@ -360,38 +376,44 @@ def run(self):  # pragma: no cover
...
+                CallbackQueryHandler(self.callback_on_add_search, pattern=Commands.add_search),
                 MessageHandler(Filters.text, self.callback_on_unknown_message),
             ],
             states={
                 States.ADD_SEARCH_ANSWER: [
                     MessageHandler(Filters.regex("^(?!/).*"), self.callback_on_do_save_search)
                 ],
                 States.STOPPED: [
-                    CallbackQueryHandler(self.callback_on_stop_feedback, pattern=r"\/stop_feedback .*"),
+                    CallbackQueryHandler(self.callback_on_stop_feedback, pattern=rf"\/{Commands.stop_feedback} .*"),
```
Comment (rooom13) [inline hot_tweets_bot/app.py:410]: porque aqui neceitamos "\/" and in line 288 no need for it?

File: hot_tweets_bot/app.py [hunk 13]
```diff
@@ -360,38 +376,44 @@ def run(self):  # pragma: no cover
...
                     MessageHandler(Filters.regex("^(?!/).*"), self.callback_on_stop_feedback),
                 ]
             },
             fallbacks=[
-                CommandHandler("cancel", self.callback_on_cancel),
-                CallbackQueryHandler(self.callback_on_cancel, pattern=r"\/cancel"),
+                CommandHandler({Commands.cancel}, self.callback_on_cancel),
+                CallbackQueryHandler(self.callback_on_cancel, pattern=rf"\/{Commands.cancel}"),
```
Comment (rooom13) [inline hot_tweets_bot/app.py:416]: same here, why \/ is needed here and not in line 388

**Review by rooom13 — CHANGES_REQUESTED** [review]

File: hot_tweets_bot/app.py (hunk 13 above)
Comment (rangelrey) [reply hot_tweets_bot/app.py:416]: typo

Comment (rangelrey) [reply hot_tweets_bot/app.py:416]: typo in the previous one

**Review by rooom13 — APPROVED** [review]

**Pull request merged**